### PR TITLE
[WIP] feat: add fromThemeUI conversion utility 

### LINKED ISSUE #23: Migration from Theme UI
## Motivation

The ecosystem behind Theme UI is growing fast. It'd be great to have a converter available for their own theme format. Also, a detailed documentation page which compares the differences between theme specs would also be very useful.

## Basic example

```js
/* theme.treat.js */

import { createTheme, fromThemeUI } from 'glaze';

export default createTheme(fromThemeUI({
  // Theme UI tokens, e.g.:
  colors: {
    text: '#000',
    background: '#fff',
    primary: '#07c',
    secondary: '#05a',
    accent: '#609',
    muted: '#f6f6f6',
  },
  fonts: {
    body: 'system-ui, sans-serif',
    heading: 'system-ui, sans-serif',
    monospace: 'Menlo, monospace',
  },
  fontWeights: {
    body: 400,
    heading: 700,
    bold: 700,
  },
  lineHeights: {
    body: 1.5,
    heading: 1.125,
  },
}));
```

This would even allow using existing Theme UI presets, e.g.:

```js
/* theme.treat.js */

import { bootstrap } from '@theme-ui/presets'
import { createTheme, fromThemeUI } from 'glaze';

export default createTheme(fromThemeUI({
  ...bootstrap,
  colors: {
    ...bootstrap.colors,
    brand: '#123456',
  },
}));
```

### PR DESCRIPTION
**Pull request by hennessyevan**
Closes #23 

This is a WIP. Ignore all the `console.log` stuff I'm using to visualize

This utility makes it easier to move from theme ui over to treat (glaze).

_Currently not all of theme-ui's theme specifications have analogies in glaze. Certain decisions are made during conversion_

The properties that cannot be converted directly:
 -  **Component Variants** - there is currently no support in treat or glaze for these so we ignore them and warn
 -  **Color Modes** - there is currently no support in glaze for color modes. See #7 
 - **Breakpoints** - ThemeUI breakpoints are an array of strings supporting any CSS length unit. Glaze uses an array of pixel based numbers. We convert `px` values directly. We then assume `em` and `rem` on the default `16px` root and convert against that assumption with a warning. All other units are dropped and an error is thrown.

### CODE REVIEW

**Comment by kripod** [comment]
Looks awesome, thank you for submitting this so quickly! If you have questions or need help with anything, feel free to ask 😊

**Comment by kripod** [comment]
I will continue reviewing this pull request tomorrow. There are some static typing issues left to be resolved, which I can help with.

**Comment by hennessyevan** [comment]
Thanks! Yeah there are some gaps in my knowledge of treat and/or theme-ui. Would appreciate the help 👍🏻

**Comment by hennessyevan** [comment]
Oh! Somehow vscode wasn't picking up these typing errors yesterday, I'll fix them 👍🏻

File: packages/glaze/src/theme.ts [hunk 1]
```diff
@@ -85,19 +85,29 @@ export function fromThemeUI(theme: ThemeUI): StaticTheme {
...
-      if (unit === 'px') {
-        return Number(value);
-      }
-      if (['rem', 'em'].includes(unit)) {
-        warnOnce(
-          `"${breakpoint}" was converted to ${value}. This could have unintended side-effects.`,
-        );
-        return Number(value) * 16;
+      const [, numberCandidate, unit] =
+        /(.*\d)([a-z]*)/.exec(
```
Comment (hennessyevan) [inline packages/glaze/src/theme.ts]: This ignores units like`%` (which is a valid `min-width` unit) and instead simply strips `%` incorrectly translating it into a pixel value

Comment (kripod) [reply packages/glaze/src/theme.ts]: Great observation. I think `/(.*\d)([a-z%]*)/` will suffice then, as it cannot be matched to any units later on.

**Comment by hennessyevan** [comment]
If you get whatever testing framework set up before this is closed I'd be happy to add some tests as well.

**Comment by kripod** [comment]
Thank you, I will review this after progressing with the documentation of near-zero config setup. Until then, please add a co-located `theme.test.ts` file with `console`-based assertions and titles as comments above each case.

**Comment by kripod** [comment]
Thanks again, this is a very important feature for onboarding developers.

**Pull request merged**